src/main/java/com/example/onlinestore/service/impl/UserServiceImpl.java

```diff
@@ -190,4 +194,17 @@ public User getUserByToken(String token) {
             return null;
         }
     }
+
+    /**
+     * 记录登录失败次数，如果超过阈值可以用于后续风控。
+     */
+    private void recordFailedLogin(String username) {
+        String key = "login:fail";
+        Long cnt = redisTemplate.opsForValue().increment(key);
+        if (cnt != null && cnt == 1) {
+            // 设置过期时间
+            redisTemplate.expire(key, 1, TimeUnit.DAYS);
+        }
+        logger.debug("记录失败登录次数 {} -> {}", username, cnt);
+    }
```

Comment:
标题: Redis 调用未做异常处理，可能影响登录流程稳健性

在登录失败路径调用 `recordFailedLogin` 时，若 Redis 暂不可用（连接超时/网络抖动），可能抛出运行时异常，影响原有认证错误处理。建议将计数逻辑降级：捕获异常并仅记录告警，不影响登录失败的正常返回。

```suggestion
    private void recordFailedLogin(String username) {
        try {
            String safeUsername = (username == null || username.isBlank()) ? "unknown" : username;
            String key = "login:fail:" + safeUsername;
            Long cnt = redisTemplate.opsForValue().increment(key);
            if (cnt != null && cnt == 1) {
                redisTemplate.expire(key, 1, TimeUnit.DAYS);
            }
            logger.debug("记录失败登录次数 [{}] -> {}", safeUsername, cnt);
        } catch (Exception e) {
            logger.warn("记录失败登录次数时发生异常: {}", e.getMessage());
        }
    }
```


---
*Powered by Qoder*